### DISCUSSION TITLE
What is the right environment variable and value to set when building for production?

### DISCUSSION THREAD

**Discussion by saikatdas0790** [leptos-rs/leptos#2011] [Q&A] [answered]
For a SSR'd `cargo-leptos` project, the `Cargo.toml` has an entry that looks like this:
![image](https://github.com/leptos-rs/leptos/assets/7412443/59a2b16b-ad7c-4598-9594-1d6ffe9938b8)

However, the deployment documentation in the book here:
https://leptos-rs.github.io/leptos/deployment.html

sets the `APP_ENVIRONMENT` variable to `production`

![image](https://github.com/leptos-rs/leptos/assets/7412443/b18eb83e-2a7a-49a5-86a3-d70128d1cd37)

I am trying to confirm the exact enviroment variable and its corresponding value to set in my production builds. Is the value "PROD" or "production"? And is the variable name documented in the book accurate?

Thoughts on the right value to be used?

**Comment by gbj** [comment]
`ENV` is correct, the Dockerfile is just built on an older version and that is out of date. I'll remove it in the docs.

**Reply by saikatdas0790** [reply]
Just to confirm, the environment variable equivalent then becomes `ENV` or `LEPTOS_ENV`?

```bash
LEPTOS_ENV="PROD"
```

Or should this environment variable be removed completely and the binary has logic inside to detect what mode it needs to run in?

**Reply by saikatdas0790** [reply]
Any thoughts on this? What's the exact environment variable to be set in the Dockerfile?

**Reply by gbj** [reply, marked as answer]
Sorry, I misread your question above.

This is a case in which I need to look through the source code to find the answer, so I'm going to do that, and paste in what I find, so you can see how I did that.

Searching in the Leptos repo for `LEPTOS_ENV` brings me to `leptos_config/src/lib.rs` and to the default initialization of a field in the definition of a struct called `LeptosOptions`
https://github.com/leptos-rs/leptos/blob/8573f22d963b82d2877df7042d2499e819234c2b/leptos_config/src/lib.rs#L96

Ok, so the struct field is named `env` and the environment variable is `LEPTOS_ENV`. 

I'm curious what the type of `env` is, so I scroll up and see that it's... `Env`
https://github.com/leptos-rs/leptos/blob/8573f22d963b82d2877df7042d2499e819234c2b/leptos_config/src/lib.rs#L47

Conveniently GitHub links me to the enum definition of `Env`, which is either `PROD` or `DEV`
https://github.com/leptos-rs/leptos/blob/8573f22d963b82d2877df7042d2499e819234c2b/leptos_config/src/lib.rs#L161-L164

Right below it, I find a function called `env_from_str`
https://github.com/leptos-rs/leptos/blob/8573f22d963b82d2877df7042d2499e819234c2b/leptos_config/src/lib.rs#L172-L182

This shows me that there are four options for the value: `"dev"` or `"development"` (same thing) and `"prod"` or `"production"` (same thing).

So the correct answer is `LEPTOS_ENV=(one of those four)`

**AND NOTE**: To the best of my knowledge, this setting is not used anywhere in Leptos itself. It exists solely in case *you* need to distinguish between dev and prod environments (as distinct from debug and release builds) anywhere in your app, and want to be able to conveniently specify that with an env variable and pull it out of the `LeptosOptions` struct that other config data is packed into.

Does that help?

**Reply by saikatdas0790** [reply]
Yes, thank you so much. That was succinct and instructive ❤️
